src/components/tools/Location/StaticMap.svelte

```diff
@@ -34,41 +31,55 @@
     };
   }
 
-  $: if (location) {
-    if (location.geometry.type === "Point") {
-      geojson = location.geometry;
-      const [lng, lat] = geojson.coordinates;
-      // Use zoom for point geometry to prevent returned image from
-      // being zoomed all the way in
-      bounds = `${lng},${lat},${zoom}`;
-      overlay = encodeURIComponent(JSON.stringify(geojson));
-      // Padding cannot be used without the auto parameter/bounding box.
-      src = `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?&access_token=${mapboxgl.accessToken}`;
+  function createPolygonImgSrc({ geometry }) {
+    // Add style for non point geometry
+    const geojson = {
+      type: "Feature",
+      properties: {
+        "fill-opacity": 0,
+        "stroke-width": 3,
+      },
+      geometry,
+    };
+    const bounds = "auto";
+    // The Static Images API only accepts requests that are 8,192 or fewer characters long. Reduce the length of the geojson overlay or request a static map without the overlay
+    let overlay;
+    let simplifiedGeojson;
+    const geojsonString = JSON.stringify(geojson);
+
+    if (geojsonString.length < 7000) {
+      overlay = encodeURIComponent(geojsonString);
     } else {
-      // Add style for non point geometry
-      geojson = {
-        type: "Feature",
-        properties: {
-          "fill-opacity": 0,
-          "stroke-width": 3,
-        },
-        geometry: location.geometry,
-      };
-      // The Static Images API only accepts requests that are 8,192 or fewer characters long.
-      // Simplify geometry only for more complex shapes
-      const geojsonString = JSON.stringify(geojson);
-      if (geojsonString.length < 7000) {
-        overlay = encodeURIComponent(geojsonString);
-      } else {
-        const simplifiedGeojson = simplify(geojson, {
-          tolerance: 0.01,
-          highQuality: true,
+      // Simplify geometry for more complex shapes
+      simplifiedGeojson = simplify(geojson, {
+        tolerance: 0.005,
+        highQuality: true,
+      });
+      // Simplify further for even more complex shapes (e.g. state boundaries)
+      if (JSON.stringify(simplifiedGeojson).length > 8000) {
```

Comment:
This seems like it might be a good use for recursion? Like recursively simplify the geometry until it's at the desired tolerance. That would catch any cases where this `if` clause passes but the geometry is still too big.

Reply:
Recursion makes my head hurt :-) but I tried it out in  https://github.com/berkeley-gif/caladapt-website-2021/pull/74/commits/60ef734801f76afb5a70bc89563e64cd045dbee6. Let me know what you think.